spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/discovery/DiscoveryClientRouteDefinitionLocator.java

```diff
@@ -122,9 +123,12 @@ private DiscoveryClientRouteDefinitionLocator(String discoveryClientName,
 				.map(instances -> instances.get(0)).filter(includePredicate)
 				.map(instance -> {
 					String serviceId = instance.getServiceId();
+					Map<String, Object> metadata = new HashMap<>();
+					metadata.putAll(instance.getMetadata());
 
 					RouteDefinition routeDefinition = new RouteDefinition();
 					routeDefinition.setId(this.routeIdPrefix + serviceId);
+					routeDefinition.setMetadata(metadata);
```

Comment:
In all of my testing, `routeDefinition.metadata` was always empty. I don't know what it was originally intended to be used for. If preferred, I can merge the `instance.metadata` with whatever the current `routeDefinition.metadata` is set as.